docs/src/gravity_wave.md

```diff
@@ -0,0 +1,46 @@
+Gravity waves have a great impact on the atmospheric circulation. They are usually generated from topography or convection, propagate upward and alter temperature and winds in the middle atmosphere, and influence tropospheric circulation through downward control. The horizontal wavelength for gravity waves ranges from several kilometers to hundreds of kilometers, which is smaller than typical GCM resolution and needs to be parameterized.
+
+## Non-orographic gravity wave
+The non-orographic gravity wave drag parameterization follows the spectra methods described in [alexander1999](@cite). The following assumptions are made for this parameterization to work:
+* The wave spectrum consists of independent monochromatic waves, and wave-wave interaction is neglected when propagation and instability is computed.
+* The gravity wave propagates vertically and conservatively to the breaking level and deposits all momentum flux into that level, as opposed to the method using saturation profile described in [lindzen1981](@cite).
+* The wave breaking criterion is derived in a hydrostatic, non-rotating frame. Including nonhydrostatic and rotation is proved to have negligible impacts.
+* The gravity wave is intermittent and the intermittency is computed as the ratio of the total long-term average source to the integral of the source momentum flux spectrum.
+
+### Spetrum of the momentum flux sources
+The source spectrum with respect to phase speed is prescribed in Eq. (17) in [alexander1999](@cite). We adapt the equation so that the spectrum writes as a combination of a wide and a narrow band as
+```math
+B_0(c) = \frac{F_{S0}(c)}{\rho_0} = sgn(c-u_0) \left( Bm\_w \exp\left[ -\left( \frac{c-c_0}{c_{w\_w}} \right) \ln{2} \right] + Bm\_n \exp\left[ -\left( \frac{c-c_0}{c_{w\_n}} \right) \ln{2} \right] \right)
+```
+where the subscript ``0`` denotes values at the source level. ``c_0`` is the phase speed with the maximum flux magnitude ``Bm``. ``c_w`` is the half-width at half-maximum of the Gaussian.  ``\_w`` and ``\_n`` represent the wide and narrow bands of the spectra.
+
+### Upward propagation and wave breaking
+Waves that are reflected will be removed from the spectrum. A wave that breaks at a level above the source will deposit all its momentum flux into that level and be removed from the spectrum.
+
+The reflection frequency is defined as
+```math
+\omega_r(z) = (\frac{N(z)^2 k^2}{k^2+\alpha^2})^{1/2}
+```	
+where ``N(z)`` is the buoyancy frequency, ``k`` is the horizontal wavenumber that corresponds to a wavelength of 300 km, ``\alpha = 1/H`` where $H$ is the scale height. ``\omega_r(z)`` is used to determine for each monochromatic wave in the spectrum, whether it will be reflected at height ``z``.
+
+The instability condition is defined as
+```math
+Q(z,c) = \frac{\overline{\rho_0}}{\rho(z)} \frac{2N(z)B_0(c)}{k[c-u(z)]^3}
+```
+``Q(z,c)`` is used to determine whether the monochromatic wave of phase speed ``c`` gets unstable at height ``z``.
```

Comment:
Define `\overline{\rho_0}`?

Reply:
It should be without \overline. Got fixed. The 0 subscript has been defined already.